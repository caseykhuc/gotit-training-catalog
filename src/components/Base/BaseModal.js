import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'
import { hideModal } from '../../actions/modal';

export const BaseModal = ({
  title, hideModal, onAccept, children,
}) => (
    <Modal show onHide={hideModal} centered>
      <Modal.Header closeButton>
        <Modal.Title>{title}</Modal.Title>
      </Modal.Header>
      <Modal.Body>{children}</Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" onClick={hideModal}>
          Close
      </Button>
        <Button variant="primary" onClick={onAccept}>
          Accept
      </Button>
      </Modal.Footer>
    </Modal>
  )

BaseModal.propTypes = {
  title: PropTypes.string.isRequired,
  hideModal: PropTypes.func.isRequired,
  onAccept: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
}

export default connect(null, { hideModal })(BaseModal);
