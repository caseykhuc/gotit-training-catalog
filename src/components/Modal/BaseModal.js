import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import { hideModal } from '../../actions';
import { connect } from 'react-redux';

class BaseModal extends React.Component {
  render() {
    return (
      <Modal.Dialog>
        <Modal.Header closeButton onClick={this.props.hideModal}>
          <Modal.Title>Modal title</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          <p>Modal body text goes here.</p>
        </Modal.Body>

        <Modal.Footer>
          <Button variant='secondary'>Close</Button>
          <Button variant='primary'>Save changes</Button>
        </Modal.Footer>
      </Modal.Dialog>
    );
  }
}

export default connect(null, { hideModal })(BaseModal);
