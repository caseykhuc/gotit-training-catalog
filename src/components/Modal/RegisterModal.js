import React from 'react';
import { Form } from 'react-bootstrap';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import BaseModal from './BaseModal';
import { registerUser } from '../../actions/user';

export class RegisterModal extends React.Component {
  state = {
    username: '',
    email: '',
    name: '',
    password: '',
  }

  onInputChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  onFormSubmit = () => {
    const { registerUser } = this.props;
    const {
      username, email, name, password,
    } = this.state;
    registerUser(username, email, name, password);
  }

  onKeyDown = (e) => {
    if (e.key === 'Enter') {
      this.onFormSubmit();
    }
  }

  render() {
    const {
      username, email, name, password,
    } = this.state;

    return (
      <div onKeyDown={(e) => this.onKeyDown(e)}>
        <BaseModal title="REGISTER" onAccept={() => this.onFormSubmit()}>
          <Form.Group controlId="username">
            <Form.Label>Username</Form.Label>
            <Form.Control type="text" name="username" value={username} placeholder="Enter username" onChange={this.onInputChange} />
          </Form.Group>

          <Form.Group controlId="email">
            <Form.Label>Email</Form.Label>
            <Form.Control type="email" name="email" value={email} placeholder="Enter email" onChange={this.onInputChange} />
          </Form.Group>

          <Form.Group controlId="name">
            <Form.Label>Name</Form.Label>
            <Form.Control type="text" name="name" value={name} placeholder="Enter name" onChange={this.onInputChange} />
          </Form.Group>

          <Form.Group controlId="password">
            <Form.Label>Password</Form.Label>
            <Form.Control type="password" name="password" value={password} placeholder="Password" onChange={this.onInputChange} />
          </Form.Group>
        </BaseModal>
      </div>
    )
  }
}

RegisterModal.propTypes = {
  registerUser: PropTypes.func.isRequired,
}

export default connect(null, { registerUser })(RegisterModal);
