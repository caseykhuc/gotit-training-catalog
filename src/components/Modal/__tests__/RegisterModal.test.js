import React from 'react';
import { shallow } from 'enzyme';
import { RegisterModal } from '../RegisterModal';
import BaseModal from '../BaseModal';

describe('components/RegisterModal', () => {
  let props;
  let wrapper;
  let container;
  let usernameField;
  let baseModal;

  const update = () => {
    wrapper.update();
    container = wrapper.find('div');
    usernameField = wrapper.find('[name="username"]');
    baseModal = wrapper.find(BaseModal);
  };
  const setup = () => {
    wrapper = shallow(<RegisterModal {...props} />);
    update();
  };
  beforeEach(() => {
    props = {
      registerUser: jest.fn(),
    }
  });
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should invoke register when user press Enter', () => {
    setup();
    container.props().onKeyDown({ key: 'Enter' });
    expect(props.registerUser).toHaveBeenCalled();
  });
  it('should not invoke register when user press other key', () => {
    setup();
    container.props().onKeyDown({ key: 'a' });
    expect(props.registerUser).not.toHaveBeenCalled();
  });
  it('should call setState on input change', () => {
    setup();
    usernameField.simulate('change', { target: { name: 'username', value: 'test' } });
    expect(wrapper.state().username).toEqual('test')
  });
  it('should invoke register when baseModal invoked onAccept ', () => {
    setup();
    baseModal.props().onAccept();
    expect(props.registerUser).toHaveBeenCalled();
  })
})
