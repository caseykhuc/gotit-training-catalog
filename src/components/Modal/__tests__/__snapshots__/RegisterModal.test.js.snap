// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RegisterModal should render correctly 1`] = `ShallowWrapper {}`;
